packages/devtools_app/test/flutter/split_test.dart

```diff
@@ -13,243 +13,511 @@ void main() {
+        await tester.drag(
+            find.byKey(split.dividerKey(1)), const Offset(-100, 0));
         await tester.pumpAndSettle();
-        expect(tester.element(find.byKey(_w1)).size, const Size(800, 0));
-        expect(tester.element(find.byKey(_w2)).size, const Size(800, 590));
+        expectEqualSizes(
+            tester.element(find.byKey(_k1)).size, const Size(370.5, 600));
+        expectEqualSizes(
+            tester.element(find.byKey(_k2)).size, const Size(0, 600));
+        expectEqualSizes(
+            tester.element(find.byKey(_k3)).size, const Size(409.5, 600));
+
+        // Drag second splitter to the right end of the widget.
+        await tester.drag(
+            find.byKey(split.dividerKey(1)), const Offset(420, 0));
+        await tester.pumpAndSettle();
+        expectEqualSizes(
+            tester.element(find.byKey(_k1)).size, const Size(370.5, 600));
+        expectEqualSizes(
+            tester.element(find.byKey(_k2)).size, const Size(409.5, 600));
+        expectEqualSizes(
+            tester.element(find.byKey(_k3)).size, const Size(0, 600));
+
+        // Make sure we can't overdrag.
+        await tester.drag(
+            find.byKey(split.dividerKey(1)), const Offset(200, 0));
+        await tester.pumpAndSettle();
+        expectEqualSizes(
+            tester.element(find.byKey(_k1)).size, const Size(370.5, 600));
+        expectEqualSizes(
+            tester.element(find.byKey(_k2)).size, const Size(409.5, 600));
+        expectEqualSizes(
+            tester.element(find.byKey(_k3)).size, const Size(0, 600));
       });
     });
 
     group('resizes contents', () {
       testWidgets('in a horizontal layout', (WidgetTester tester) async {
-        final split = buildSplit(Axis.horizontal, initialFirstFraction: 0.0);
+        final split = buildSplit(Axis.horizontal, initialFractions: [0.0, 1.0]);
         await tester.pumpWidget(wrap(
           Center(
             child: SizedBox(width: 300.0, height: 300.0, child: split),
           ),
         ));
-        expect(tester.element(find.byKey(_w1)).size, const Size(0, 300));
-        expect(tester.element(find.byKey(_w2)).size, const Size(290, 300));
+        expectEqualSizes(
+            tester.element(find.byKey(_k1)).size, const Size(0, 300));
+        expectEqualSizes(
+            tester.element(find.byKey(_k2)).size, const Size(290, 300));
 
         await tester.pumpWidget(wrap(
           Center(
             child: SizedBox(width: 200.0, height: 200.0, child: split),
           ),
         ));
-        expect(tester.element(find.byKey(_w1)).size, const Size(0, 200));
-        expect(tester.element(find.byKey(_w2)).size, const Size(190, 200));
+        expectEqualSizes(
+            tester.element(find.byKey(_k1)).size, const Size(0, 200));
+        expectEqualSizes(
+            tester.element(find.byKey(_k2)).size, const Size(190, 200));
+      });
+
+      testWidgets('in a horizontal layout with n children',
```

Comment:
add a test with epsilon rounding errors so initialFractions add up to 0.99999999

Reply:
done